### PR TITLE
Channel materials example: normal, depth, depthRGBA

### PR DESCRIPTION
**Pull request by bhouston**
This is a new example that allows one to see the various packed channels, normals, depth, depthRGBA, and depthRGBADebug (which is a decoded depthRGBA channel.)  It also lets you see try both the perspective and normal cameras.

In this PR it seems that the MeshDepthMaterial may not actually work.  It is confusing to be what is going on with it, or its intended purpose.  I'll submit that as an issue so that it is separate from this PR.

### CODE REVIEW

**Comment by bhouston** [comment]
Here is a link to the new example:

http://ci.threejs.org/api/pullrequests/8549/examples/webgl_materials_channels.html

image

image

**Comment by WestLangley** [comment]
Very informative example. : - )

>And these two materials do not completely match up with the MeshStandardMaterial when it has a normal map and a displacement map:

Not sure what you are referring to...

**Comment by bhouston** [comment]
>> And these two materials do not completely match up with the MeshStandardMaterial when it has a normal map and a displacement map:
> Not sure what you are referring to...

The normal channel doesn't contain the normals from the normal map used by the standard material, thus in my opinion the normal channel is not fully complete.  Thus my upcoming SAO shader will not capture microstructure detail represented in the normal map.

The depth and normal channels do not align with the standard material render because depth and normal shaders do not incorporate the displacement map that shifts around vertices.  Thus shadows and my upcoming SAO shader will not be usable with displaced geometry as its results will not align with these meshes at all (and in fact the existing SSAO and DOF shaders also are incompatible with displaced geometry.)

**Comment by WestLangley** [comment]
I agree. We should fix that.

**Comment by mrdoob** [comment]
Seems like this PR needs to be rebased.

**Comment by bhouston** [comment]
> I agree. We should fix that.

I was thinking we should but I think it is very inefficient.  I am not sure it is worth it because doing normals and depths as separate render passes is already slow, and this will make it slower.

The cool thing is that with WebGL 2.0 we will be able to just output the normals and depths along with the colors in a single pass -- thus fast and complete.  So I am a little okay with reduced quality for now.  I can get SAO working and then with WebGL 2.0 we can make it more beautiful and more fast.

> Seems like this PR needs to be rebased.

It is caused by collapsing the change sets.  The issue with collapsing the change sets it it is hard to find the original individual commits (which I really love for tracking down bugs, the descriptions per commit are amazing) or find the PR numbers of what was merged (I also miss the "Merge pull request #8488 from Mugen87/dev" descriptions.)  I agree it leads to a shorter history but I find it is a harder to follow history.  I will rebase this though.

**Comment by bhouston** [comment]
This PR has been rebased and polished a bit more - should be ready to merge:

http://ci.threejs.org/api/pullrequests/8549/examples/webgl_materials_channels.html

**Comment by WestLangley** [comment]
@bhouston Look nice! BTW, the depth channel renders non-white if you move the camera back so the object is in the far end of the camera frustum.

**Comment by bhouston** [comment]
@WestLangley I fixed the depth shader with this other PR: #8558.  I am pretty sure the current implementation is just wrong and it also wouldn't work with orthographic cameras. :/

File: examples/files.js [hunk 1]
```diff
@@ -134,6 +135,7 @@ var files = {
 		"webgl_materials_reflectivity",
 		"webgl_materials_shaders_fresnel",
 		"webgl_materials_skin",
+		"webgl_materials_channels",
```
Comment (mrdoob) [inline examples/files.js]: Seems like you've added the example to the list twice.

Comment (bhouston) [inline examples/files.js]: fixed. :)

**Comment by mrdoob** [comment]
Thanks thanks!

**Pull request merged**